src/turbomind/windows/CMakeLists.txt

```diff
@@ -0,0 +1,24 @@
+cmake_minimum_required(VERSION 3.14)
+
+if (NOT MSVC)
+    return ()
+endif ()
+
+include(FetchContent)
+
+# libpthreadVC3
+SET(PTHREAD_WIN32_SRC ${CMAKE_BINARY_DIR}/_deps/pthread-win32-src/)
+SET(PTHREAD_WIN32_BINARY ${CMAKE_BINARY_DIR}/_deps/pthread-win32-build/)
+set(CMAKE_MODULE_PATH "${CMAKE_MODULE_PATH}" "${PTHREAD_WIN32_SRC}/cmake")
+
+FetchContent_Declare(
```

Comment:
We don't support TP on Windows so the use of `pthread_barrier` is trival (count == 1)

Instead of introducing whole pthread dependency, guard the pthread `Barrier` implementation with `#ifdef` and add a trival `Barrier` implementation for MSVC which simply asserts `count == 1` in ctor and does nothing in other member functions.